torch/distributed/tensor/_utils.py

```diff
@@ -15,6 +16,65 @@
 )
 
 
+def _explicit_order_placements(
+    mesh_shape: ShapeType, placements: Sequence[Placement]
```

Comment:
several util functions including this one can take `DTensorSpec` as input instead of `mesh` + `placements`. I suggest we create a task for this. WDYT?

Reply:
Ok with me, though one reason I avoided passing mesh and passed mesh shape instead is that I could more easily write a unit test on one process without creating a device mesh. I would like to keep that property somehow.